src/CurrencyFormat.php

```diff
@@ -26,19 +26,48 @@ class CurrencyFormat implements CurrencyFormatInterface
      * @var int
      */
     private $precision;
+    /**
+     * @var int
+     */
+    private $decorationSpace;
 
-    public function __construct(
+    private function __construct(
         string $decimalsSeparator = '.',
         string $thousandsSeparator = '',
-        int $extraPrecision = 0,
         int $decorationType = self::DECORATION_SYMBOL,
+        int $decorationSpace = self::DECORATION_WITHOUT_SPACE,
+        int $extraPrecision = 0,
         int $precision = null
     ) {
         $this->decimalsSeparator = $decimalsSeparator;
         $this->thousandsSeparator = $thousandsSeparator;
         $this->extraPrecision = $extraPrecision;
         $this->decorationType = $decorationType;
         $this->precision = $precision;
+        $this->decorationSpace = $decorationSpace;
+    }
+
+    public static function defaultFormatting(): CurrencyFormatInterface
+    {
+        return new self();
+    }
+
+    public static function fromParameters(
+        string $decimalsSeparator = '.',
+        string $thousandsSeparator = '',
+        int $decorationType = self::DECORATION_SYMBOL,
+        int $decorationSpace = self::DECORATION_WITHOUT_SPACE,
+        int $extraPrecision = 0,
+        int $precision = null
+    ): CurrencyFormatInterface {
```

Comment:
This static named constructor allows inconsistent formatting options (`$extraPrecision` vs `$precision`)